src/views/Project/Details/General/index.js

```diff
@@ -1,65 +1,269 @@
+                    if (pristine || location.pathname === routeUrl) {
+                        return true;
+                    }
+
+                    return _ts('common', 'youHaveUnsavedChanges');
+                }
+            }
+        />
+    )
+
     render() {
         const {
-            pending,
+            projectLocalData: {
+                faramValues = {},
+                faramErrors,
+                pristine,
+            },
         } = this.props;
 
+        const {
+            pendingProjectGet,
+            pendingProjectPut,
+        } = this.state;
+
+        const loading = pendingProjectGet || pendingProjectPut;
+        const UnsavedChangesPrompt = this.renderUnsavedChangesPrompt;
+
         return (
-            <div className={styles.projectGeneral}>
-                { pending && <LoadingAnimation /> }
-                <div
-                    className={styles.inputsContainer}
+            <React.Fragment>
+                <Faram
+                    className={styles.projectGeneral}
+                    onChange={this.handleFaramChange}
+                    onValidationFailure={this.handleValidationFailure}
+                    onValidationSuccess={this.handleValidationSuccess}
+                    schema={this.schema}
+                    value={faramValues}
+                    error={faramErrors}
+                    disabled={loading}
                 >
-                    <TextInput
-                        label={_ts('project.general', 'projectNameLabel')}
-                        faramElementName="title"
-                        placeholder={_ts('project.general', 'projectNamePlaceholder')}
-                        className={styles.name}
-                    />
-                    <DateInput
-                        label={_ts('project.general', 'projectStartDateLabel')}
-                        faramElementName="startDate"
-                        placeholder={_ts('project.general', 'projectStartDatePlaceholder')}
-                        className={styles.startDate}
-                    />
-                    <DateInput
-                        label={_ts('project.general', 'projectEndDateLabel')}
-                        faramElementName="endDate"
-                        placeholder={_ts('project.general', 'projectEndDatePlaceholder')}
-                        className={styles.endDate}
-                    />
-                </div>
-                <TextArea
-                    label={_ts('project.general', 'projectDescriptionLabel')}
-                    faramElementName="description"
-                    placeholder={_ts('project.general', 'projectDescriptionPlaceholder')}
-                    className={styles.description}
-                    rows={3}
-                />
-            </div>
+                    { loading && <LoadingAnimation /> }
+                    <div className={styles.visualizations}>
+                        Visualizations
```

Comment:
Use strings or write fixme

Reply:
Just a placeholder, Not meant to be there (Robot detected)

Reply:
If just a placeholder you should write FIXME / TODO.
*beep boop beep boop*